### ISSUE TITLE
Warning: Refs Must Have Owner

### ISSUE THREAD

**Issue by hundsim** [Fauntleroy/react-simple-dropdown#41]
I'm getting the following error:
> Uncaught Error: addComponentAsRefTo(...): Only a ReactOwner can have refs. You might be adding a ref to a component that was not created inside a component's `render` method, or you have multiple copies of React loaded 

I just installed `react-simple-dropdown` and used the exact code from the example page (except for the `props`) http://fauntleroy.github.io/react-simple-dropdown/

Is the problem on my end or has this happened to someone else?

I'm on React 15.5.4, Webpack 2.5.1 and Yarn 0.23.4

**Comment by lawrencesong** [comment]
I have similar problem. @hundsim Have you figured out a solution to remove the warning?

**Comment by Fauntleroy** [comment]
@lawrencesong what versions of `react` and `react-simple-dropdown` are you using? What does your usage of `Dropdown` and its subcomponents look like?

**Comment by lawrencesong** [comment]
@Fauntleroy I am using `react 15.6.1` and `react-simple-dropdown 3.0.0`. My usage is very similar to your example. 

I got the warning only if I put `react-simple-dropdown` in my components library and use it in my app. When I use`react-simple-dropdown` directly in my app, the warning goes away.

**Comment by Fauntleroy** [comment]
What do you mean by components library? What does putting this into the library entail?

**Comment by lawrencesong** [comment]
@Fauntleroy I created a component library to use in multiple projects. If I put `react-simple-dropdown` into `my components library`, I will see the warning when I used the `my component library` in `my app`. Dependencies look like: `my app` -> `my components library` -> `react-simple-dropdown`

If I used `react-simple-dropdown` directly in  `my app`, I won't see the warning. Dependencies look like:  `my app` -> `react-simple-dropdown`

**Comment by DonGiulio** [comment]
I'm seeing this as an error after upgrading npm to version 5.7.1 from 5.6.x.

With the previous version I didn't even get the warning, but now it's breaking my whole site. 

This is the markup for my menu component, the line generating the problem is the first one: 

```
 <Dropdown className="account-menu" ref="menu">
        <DropdownTrigger onClick={this.handleMenuClick}>Menu</DropdownTrigger>
        <DropdownContent>
          {currentUser.isSignedIn && this.signedInMenu()}
          {!currentUser.isSignedIn && this.signedOutMenu()}
        </DropdownContent>
      </Dropdown>
```

I see the error also if I remove the `ref="menu"` prop.

this is the complete error from the console: 
```
VM29719 bundle.js:33311 Uncaught Error: Element ref was specified as a string (trigger) but no owner was set. You may have multiple copies of React loaded. (details: https://fb.me/react-refs-must-have-owner).
    at invariant (VM29719 bundle.js:33311)
    at coerceRef (VM29719 bundle.js:58132)
    at createChild (VM29719 bundle.js:58373)
    at reconcileChildrenArray (VM29719 bundle.js:58678)
    at reconcileChildFibers (VM29719 bundle.js:59047)
    at reconcileChildrenAtExpirationTime (VM29719 bundle.js:59152)
    at reconcileChildren (VM29719 bundle.js:59143)
    at updateHostComponent (VM29719 bundle.js:59394)
    at beginWork (VM29719 bundle.js:59625)
    at performUnitOfWork (VM29719 bundle.js:61620)
    at workLoop (VM29719 bundle.js:61684)
    at HTMLUnknownElement.callCallback (VM29719 bundle.js:51938)
    at Object.invokeGuardedCallbackDev (VM29719 bundle.js:51977)
    at invokeGuardedCallback (VM29719 bundle.js:51834)
    at renderRoot (VM29719 bundle.js:61762)
    at performWorkOnRoot (VM29719 bundle.js:62410)
    at performWork (VM29719 bundle.js:62363)
    at requestWork (VM29719 bundle.js:62274)
    at scheduleWorkImpl (VM29719 bundle.js:62128)
    at scheduleWork (VM29719 bundle.js:62085)
    at scheduleTopLevelUpdate (VM29719 bundle.js:62589)
    at Object.updateContainer (VM29719 bundle.js:62627)
    at VM29719 bundle.js:66622
    at Object.unbatchedUpdates (VM29719 bundle.js:62498)
    at renderSubtreeIntoContainer (VM29719 bundle.js:66621)
    at Object.render (VM29719 bundle.js:66686)
    at Object../src/index.js (VM29719 bundle.js:106272)
    at __webpack_require__ (VM29719 bundle.js:679)
    at fn (VM29719 bundle.js:89)
    at Object.0 (VM29719 bundle.js:107050)
    at __webpack_require__ (VM29719 bundle.js:679)
    at ../node_modules/classnames/index.js.hasOwn (VM29719 bundle.js:725)
    at VM29719 bundle.js:728
invariant @ VM29719 bundle.js:33311
coerceRef @ VM29719 bundle.js:58132
createChild @ VM29719 bundle.js:58373
reconcileChildrenArray @ VM29719 bundle.js:58678
reconcileChildFibers @ VM29719 bundle.js:59047
reconcileChildrenAtExpirationTime @ VM29719 bundle.js:59152
reconcileChildren @ VM29719 bundle.js:59143
updateHostComponent @ VM29719 bundle.js:59394
beginWork @ VM29719 bundle.js:59625
performUnitOfWork @ VM29719 bundle.js:61620
workLoop @ VM29719 bundle.js:61684
callCallback @ VM29719 bundle.js:51938
invokeGuardedCallbackDev @ VM29719 bundle.js:51977
invokeGuardedCallback @ VM29719 bundle.js:51834
renderRoot @ VM29719 bundle.js:61762
performWorkOnRoot @ VM29719 bundle.js:62410
performWork @ VM29719 bundle.js:62363
requestWork @ VM29719 bundle.js:62274
scheduleWorkImpl @ VM29719 bundle.js:62128
scheduleWork @ VM29719 bundle.js:62085
scheduleTopLevelUpdate @ VM29719 bundle.js:62589
updateContainer @ VM29719 bundle.js:62627
(anonymous) @ VM29719 bundle.js:66622
unbatchedUpdates @ VM29719 bundle.js:62498
renderSubtreeIntoContainer @ VM29719 bundle.js:66621
render @ VM29719 bundle.js:66686
./src/index.js @ VM29719 bundle.js:106272
__webpack_require__ @ VM29719 bundle.js:679
fn @ VM29719 bundle.js:89
0 @ VM29719 bundle.js:107050
__webpack_require__ @ VM29719 bundle.js:679
../node_modules/classnames/index.js.hasOwn @ VM29719 bundle.js:725
(anonymous) @ VM29719 bundle.js:728
VM29719 bundle.js:69031 The above error occurred in the <div> component:
    in div
    in Dropdown (at Menu.jsx:144)
    in Menu (created by Connect(Menu))
    in Connect(Menu) (at App.jsx:29)
    in div (created by styled.div)
    in styled.div (at App.jsx:28)
    in div (at App.jsx:26)
    in App (created by Connect(App))
    in Connect(App) (created by Route)
    in Route (at MyApp.jsx:22)
    in Switch (at MyApp.jsx:21)
    in ThemeProvider (at MyApp.jsx:20)
    in Router (created by BrowserRouter)
    in BrowserRouter (at MyApp.jsx:19)
    in Provider (at MyApp.jsx:18)
    in MyApp (at index.js:7)
```

**Comment by supukarmin** [comment]
@lawrencesong Could you solve the issue? Facing the same problem right now.

**Comment by tonydiaz** [comment]
Is anyone getting past this issue? I can't render component due to this error. Since a ref is part of this component I can't change it.

Code: https://github.com/Fauntleroy/react-simple-dropdown/blob/c43deb49e83197767b96938939bf0ecd4208bba6/src/components/Dropdown.jsx#L89

Error:
```
Uncaught Error: Element ref was specified as a string (trigger) but no owner was set. This could happen for one of the following reasons:
1. You may be adding a ref to a functional component
2. You may be adding a ref to a component that was not created inside a component's render method
3. You have multiple copies of React loaded
See https://fb.me/react-refs-must-have-owner for more information.
    at invariant (invariant.js:42)
    at coerceRef (react-dom.development.js:7350)
    at createChild (react-dom.development.js:7555)
    at reconcileChildrenArray (react-dom.development.js:7805)
    at reconcileChildFibers (react-dom.development.js:8121)
    at reconcileChildrenAtExpirationTime (react-dom.development.js:8240)
    at reconcileChildren (react-dom.development.js:8231)
    at updateHostComponent (react-dom.development.js:8539)
    at beginWork (react-dom.development.js:8986)
    at performUnitOfWork (react-dom.development.js:11814)
```

**Comment by Fauntleroy** [comment]
This issue should be resolved with version 3.2.1

**Comment by nkov** [comment]
@Fauntleroy FYI, the issue is still present in 3.2.3, but the workaround by @tonydiaz fixes it.
